### PR TITLE
upgrade to react 18 and next 13

### PR DESCRIPTION
**Pull request by MaxMillington**
## What is the purpose of the change

Upgrading React, Next,

### ClickUp Task

[ClickUp Task URL](https://app.clickup.com/t/8678bgfru)

## Brief Changelog

Update React, Next

## Testing and Verifying

This change has been tested locally by rebuilding the website and verified content and links are expected

### CODE REVIEW

**Comment by jonator** [comment]
@MaxMillington 

Notice any improvements with the Rust-based webpack replacement?

https://nextjs.org/blog/next-13

**Comment by MaxMillington** [comment]
> @MaxMillington
> 
> Notice any improvements with the Rust-based webpack replacement?
> 
> https://nextjs.org/blog/next-13

Haven't noticed any improvements yet. I got a few errors but I've been having a hard time determining which if any are caused by the upgrade, that's why I want to try to fix those errors we are seeing in the console so I can isolate any errors that I missed.

File: packages/math/package.json [hunk 1]
```diff
@@ -16,7 +16,6 @@
     "math"
   ],
   "license": "Apache-2.0",
-  "private": false,
```
Comment (MaxMillington) [inline packages/math/package.json:19]: This was done automatically by a codemod. Do we want private? I assume so?

File: packages/web/components/cards/bond-card.tsx [hunk 2]
```diff
@@ -355,15 +355,23 @@ const Drawer: FunctionComponent<{
...
-  undelegating,
-  validatorMoniker,
-  validatorLogoUrl,
-}) => {
+const SuperfluidBreakdownRow: FunctionComponent<BondDuration["superfluid"]> = (
+  props
+) => {
   const t = useTranslation();
+
+  if (!props) {
```
Comment (MaxMillington) [inline packages/web/components/cards/bond-card.tsx:363]: all props were potentially undefined

File: packages/web/components/cards/pool-card.tsx [hunk 3]
```diff
@@ -25,48 +25,49 @@ export const PoolCard: FunctionComponent<
   const t = useTranslation();
 
   return (
-    <Link href={`/pool/${poolId}`} passHref prefetch={false}>
-      <a
```
Comment (MaxMillington) [inline packages/web/components/cards/pool-card.tsx:29]: <a> tags can no longer be children of <Link>

File: packages/web/components/client-only.tsx [hunk 4]
```diff
@@ -1,6 +1,8 @@
 import { FunctionComponent, useEffect, useState } from "react";
 
-const ClientOnly: FunctionComponent<{ className?: string }> = (props) => {
+const ClientOnly: FunctionComponent<{ className?: string; children?: any }> = (
```
Comment (MaxMillington) [inline packages/web/components/client-only.tsx:3]: `children` must be explicitly typed now

File: packages/web/components/complex/add-liquidity.tsx [hunk 5]
```diff
@@ -228,7 +228,7 @@ export const AddLiquidity: FunctionComponent<
               </Tooltip>{" "}
               {t("addLiquidity.scalingFactorControllerWarning_third")}{" "}
               {addLiquidityConfig.stableSwapInfo?.scalingFactor?.map(
-                (factor, index, array) => {
+                (factor: any, index: number, array: string | any[]) => {
```
Comment (MaxMillington) [inline packages/web/components/complex/add-liquidity.tsx:231]: implicit `any` is no longer allowed

Comment (jonator) [reply packages/web/components/complex/add-liquidity.tsx:231]: These should have types.

**Comment by jonator** [comment]
@MaxMillington in the name of cleaning up PRs, would you consider this one stale?

File: packages/web/package.json [hunk 6]
```diff
@@ -75,13 +75,13 @@
...
     "mobx-react-lite": "^3.2.2",
     "mobx-utils": "^6.0.4",
-    "next": "12.0.10",
+    "next": "13.4.9",
     "next-seo": "^6.0.0",
     "qrcode.react": "^1.0.1",
-    "react": "17.0.2",
-    "react-dom": "17.0.2",
+    "react": "^18.0.0",
+    "react-dom": "^18.0.0",
```
Comment (JoseRFelix) [inline packages/web/package.json:82]: [Next.js min react version](https://nextjs.org/docs/pages/building-your-application/upgrading/version-13) is `18.2.0`

```suggestion
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
```

File: packages/web/package.json [hunk 7]
```diff
@@ -25,12 +25,12 @@
     "@cosmjs/cosmwasm-stargate": "0.29.0",
     "@cosmjs/proto-signing": "0.29.0",
     "@cosmjs/stargate": "0.29.0",
-    "@cosmos-kit/core": "1.7.0",
+    "@cosmos-kit/core": "latest",
```
Comment (JoseRFelix) [inline packages/web/package.json:28]: Let's keep `@comsos-kit/core` to a locked version

**Comment by MaxMillington** [comment]
@jonator No, let's keep this one open. We need to figure this out as every month we put it off it gets more difficult. A substantial portion of the work is just fixing all the type errors in our repo, and this PR fixes quite a bit of them. 

That said, I am not going to deprioritize staking work for this nor should we deprioritize token info pages

**Pull request closed without merging**